{
  "type": "Program",
  "body": [
    {
      "type": "TryStatement",
      "block": {
        "type": "BlockStatement",
        "body": [
          {
            "type": "ThrowStatement",
            "argument": {
              "type": "Literal",
              "value": "catchme",
              "raw": "\"catchme\"",
              "start": 310,
              "end": 319
            },
            "start": 304,
            "end": 320
          },
          {
            "type": "ThrowStatement",
            "argument": {
              "type": "NewExpression",
              "callee": {
                "type": "Identifier",
                "name": "Test262Error",
                "start": 333,
                "end": 345
              },
              "arguments": [
                {
                  "type": "Literal",
                  "value": "#1.1: throw \"catchme\" lead to throwing exception",
                  "raw": "'#1.1: throw \"catchme\" lead to throwing exception'",
                  "start": 346,
                  "end": 396
                }
              ],
              "start": 329,
              "end": 397
            },
            "start": 323,
            "end": 398
          }
        ],
        "start": 300,
        "end": 400
      },
      "handler": {
        "type": "CatchClause",
        "param": {
          "type": "Identifier",
          "name": "e",
          "start": 408,
          "end": 409
        },
        "body": {
          "type": "BlockStatement",
          "body": [
            {
              "type": "IfStatement",
              "test": {
                "type": "UnaryExpression",
                "operator": "delete",
                "argument": {
                  "type": "Identifier",
                  "name": "e",
                  "start": 426,
                  "end": 427
                },
                "prefix": true,
                "start": 419,
                "end": 427
              },
              "consequent": {
                "type": "BlockStatement",
                "body": [
                  {
                    "type": "ThrowStatement",
                    "argument": {
                      "type": "NewExpression",
                      "callee": {
                        "type": "Identifier",
                        "name": "Test262Error",
                        "start": 444,
                        "end": 456
                      },
                      "arguments": [
                        {
                          "type": "Literal",
                          "value": "#1.2: Exception has DontDelete property",
                          "raw": "'#1.2: Exception has DontDelete property'",
                          "start": 457,
                          "end": 498
                        }
                      ],
                      "start": 440,
                      "end": 499
                    },
                    "start": 434,
                    "end": 500
                  }
                ],
                "start": 428,
                "end": 504
              },
              "alternate": null,
              "start": 415,
              "end": 504
            },
            {
              "type": "IfStatement",
              "test": {
                "type": "BinaryExpression",
                "left": {
                  "type": "Identifier",
                  "name": "e",
                  "start": 511,
                  "end": 512
                },
                "operator": "!==",
                "right": {
                  "type": "Literal",
                  "value": "catchme",
                  "raw": "\"catchme\"",
                  "start": 515,
                  "end": 524
                },
                "start": 511,
                "end": 524
              },
              "consequent": {
                "type": "BlockStatement",
                "body": [
                  {
                    "type": "ThrowStatement",
                    "argument": {
                      "type": "NewExpression",
                      "callee": {
                        "type": "Identifier",
                        "name": "Test262Error",
                        "start": 542,
                        "end": 554
                      },
                      "arguments": [
                        {
                          "type": "BinaryExpression",
                          "left": {
                            "type": "Literal",
                            "value": "#1.3: Exception === \"catchme\". Actual:  Exception ===",
                            "raw": "'#1.3: Exception === \"catchme\". Actual:  Exception ==='",
                            "start": 555,
                            "end": 610
                          },
                          "operator": "+",
                          "right": {
                            "type": "Identifier",
                            "name": "e",
                            "start": 612,
                            "end": 613
                          },
                          "start": 555,
                          "end": 613
                        }
                      ],
                      "start": 538,
                      "end": 616
                    },
                    "start": 532,
                    "end": 617
                  }
                ],
                "start": 526,
                "end": 621
              },
              "alternate": null,
              "start": 507,
              "end": 621
            }
          ],
          "start": 411,
          "end": 623
        },
        "start": 401,
        "end": 623
      },
      "finalizer": null,
      "start": 296,
      "end": 623
    },
    {
      "type": "TryStatement",
      "block": {
        "type": "BlockStatement",
        "body": [
          {
            "type": "ThrowStatement",
            "argument": {
              "type": "Literal",
              "value": "catchme",
              "raw": "\"catchme\"",
              "start": 650,
              "end": 659
            },
            "start": 644,
            "end": 660
          },
          {
            "type": "ThrowStatement",
            "argument": {
              "type": "NewExpression",
              "callee": {
                "type": "Identifier",
                "name": "Test262Error",
                "start": 673,
                "end": 685
              },
              "arguments": [
                {
                  "type": "Literal",
                  "value": "#2.1: throw \"catchme\" lead to throwing exception",
                  "raw": "'#2.1: throw \"catchme\" lead to throwing exception'",
                  "start": 686,
                  "end": 736
                }
              ],
              "start": 669,
              "end": 737
            },
            "start": 663,
            "end": 738
          }
        ],
        "start": 640,
        "end": 740
      },
      "handler": {
        "type": "CatchClause",
        "param": {
          "type": "Identifier",
          "name": "e",
          "start": 747,
          "end": 748
        },
        "body": {
          "type": "BlockStatement",
          "body": [],
          "start": 749,
          "end": 751
        },
        "start": 741,
        "end": 751
      },
      "finalizer": null,
      "start": 636,
      "end": 751
    },
    {
      "type": "TryStatement",
      "block": {
        "type": "BlockStatement",
        "body": [
          {
            "type": "ExpressionStatement",
            "expression": {
              "type": "Identifier",
              "name": "e",
              "start": 759,
              "end": 760
            },
            "start": 759,
            "end": 761
          },
          {
            "type": "ThrowStatement",
            "argument": {
              "type": "NewExpression",
              "callee": {
                "type": "Identifier",
                "name": "Test262Error",
                "start": 774,
                "end": 786
              },
              "arguments": [
                {
                  "type": "Literal",
                  "value": "#2.2: Deleting catching exception after ending \"catch\" block",
                  "raw": "'#2.2: Deleting catching exception after ending \"catch\" block'",
                  "start": 787,
                  "end": 849
                }
              ],
              "start": 770,
              "end": 850
            },
            "start": 764,
            "end": 851
          }
        ],
        "start": 755,
        "end": 853
      },
      "handler": {
        "type": "CatchClause",
        "param": {
          "type": "Identifier",
          "name": "err",
          "start": 860,
          "end": 863
        },
        "body": {
          "type": "BlockStatement",
          "body": [],
          "start": 864,
          "end": 866
        },
        "start": 854,
        "end": 866
      },
      "finalizer": null,
      "start": 752,
      "end": 866
    }
  ],
  "sourceType": "script",
  "hashbang": null,
  "start": 0,
  "end": 867
}